app/workers/tulsa_city_worker.rb

```diff
@@ -0,0 +1,88 @@
+require 'byebug'
+require 'json'
+
+class TulsaCityWorker
+  include Sidekiq::Worker
+  include Sidekiq::Throttled::Worker
+  sidekiq_options retry: 5, queue: :high
+  sidekiq_throttle_as :tulsa_city
+  def self.perform(recordid, active)
+    new.perform(recordid, active)
+  end
+
+  def perform(incidentrecordid, active)
+    inmate = perform_inmate(incidentrecordid)
+
+    inmate = inmate.transform_values(&:strip)
+    tulsa_inmate = save_inmate(inmate, incidentrecordid, active)
+
+    offense_json = perform_offense(tulsa_inmate.incident_record_id)
+    offense_json.each do |offense|
+      offense = offense.transform_values(&:strip)
+      save_offense(offense, tulsa_inmate)
+    end
+  end
+
+  def save_inmate(inmate_json, incidentrecordid, active)
+    inmate = ::TulsaCity::Inmate.find_or_create_by(incident_record_id: incidentrecordid)
+    inmate.inmate_id = inmate_json['inmateId']
+    inmate.first_name = inmate_json['firstName']
+    inmate.middle_name = inmate_json['middleName']
+    inmate.last_name = inmate_json['lastName']
+    inmate.height = inmate_json['height']
+    inmate.weight = inmate_json['weight']
+    inmate.hair_color = inmate_json['hairColor']
+    inmate.eye_color = inmate_json['eyeColor']
+    inmate.race = inmate_json['race']
+    inmate.gender = inmate_json['gender']
+    inmate.arrest_date = inmate_json['arrestDate']
+    inmate.arresting_officer = inmate_json['arrestingOfficer']
+    inmate.arresting_agency = inmate_json['arrestingAgency']
+    inmate.booking_date_time = inmate_json['bookingDateTime']
+    inmate.court_date = inmate_json['CourtDate']
+    inmate.released_date_time = inmate_json['releasedDateTime']
+    inmate.court_division = inmate_json['courtDivision']
+    inmate.incident_record_id = incidentrecordid
+    inmate.dob = inmate_json['DOB']
+    inmate.active = active
+    inmate.save!
+    inmate
+  end
+
+  def save_offense(offense_json, inmate)
+    offense = ::TulsaCity::Offense.find_or_create_by(docket_id: offense_json['docketId'])
```

Comment:
This might be okay but I'm actually not 100% sure that multiple inmates couldn't have the same docketId if they were part of the same case
would be safer here to also do it by inmate_id

Reply:
`offense = ::TulsaCity::Offense.find_or_create_by(docket_id: offense_json['docketId'], inmate_id: inmate.id)`